examples/coexist_klog_v1_and_v2/coexist_klog_v1_and_v2.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"flag"
+
+	klogv1 "k8s.io/klog"
+	klogv2 "k8s.io/klog/v2"
+)
+
+// OutputCallDepth is the stack depth where we can find the origin of this call
+const OutputCallDepth = 6
+// log prefix that we have to strip out
+const DefaultPrefixLength = 53
+
+// klogWriter is used in SetOutputBySeverity call below to redirect
+// any calls to klogv1 to end up in klogv2
+type klogWriter struct{}
+
+func (kw klogWriter) Write(p []byte) (n int, err error) {
+	if len(p) < DefaultPrefixLength {
+		klogv2.InfoDepth(OutputCallDepth, string(p))
+		return len(p), nil
+	}
+	if p[0] == 'I' {
+		klogv2.InfoDepth(OutputCallDepth, string(p[DefaultPrefixLength:]))
+	} else if p[0] == 'W' {
+		klogv2.WarningDepth(OutputCallDepth, string(p[DefaultPrefixLength:]))
+	} else if p[0] == 'E' {
+		klogv2.ErrorDepth(OutputCallDepth, string(p[DefaultPrefixLength:]))
+	} else if p[0] == 'F' {
+		klogv2.FatalDepth(OutputCallDepth, string(p[DefaultPrefixLength:]))
+	} else {
+		klogv2.InfoDepth(OutputCallDepth, string(p[DefaultPrefixLength:]))
+	}
+	return len(p) - DefaultPrefixLength, nil
+}
+
+func main() {
+	// Assume that we have to switch klogv2 on from now on everywhere.
+	klogv2.InitFlags(nil)
+
+	// These following are just things you do to ensure that you stream
+	// all the things to the log file instead of stdout/stderr
+	flag.Set("logtostderr", "false")     // By default klog logs to stderr, switch that off
+	flag.Set("alsologtostderr", "false") // false is default, but this is informative
+	flag.Set("stderrthreshold", "FATAL") // stderrthreshold defaults to ERROR, we don't want anything in stderr
+	flag.Set("log_file", "myfile.log")   // log to a file
+	flag.Parse()
```

Comment:
are all of these options required for this example? I thought we were just 

can we simplify the klog/v2 portion to:

```go
// initialize klog/v2, can also bind to a local flagset if desired
klogv2.InitFlags(nil)
// ... add any other desired klog/v2 setup here
// parse klog/v2 flags
flag.Parse()
// make sure we flush before exiting
defer klogv2.Flush()
```

Reply:
I'd like to keep them for illustration purposes when folks try out the example, they should see nothing in stderr/out and all the log lines should be in the myfile.log, i'll try to word the comments better.